src/main/java/com/aws/greengrass/secretmanager/crypto/RSAMasterKey.java

```diff
@@ -14,18 +17,34 @@
 import java.security.PrivateKey;
 import java.security.PublicKey;
 
+import static com.aws.greengrass.secretmanager.SecretManagerService.SECRET_MANAGER_SERVICE_NAME;
+
 public class RSAMasterKey implements MasterKey {
+    private static final Logger logger = LogManager.getLogger(RSAMasterKey.class);
     // Metadata used by JCEMasterKey, so that another instance of JCEMasterKey cannot be used
     // to encrypt/decrypt payload even if it uses the same key.
     private static final String KEY_PROVIDER = "gg:secrets";
-    private static final String WRAPPING_ALGO = "RSA/ECB/OAEPWithSHA-256AndMGF1Padding";
+    private static final String[] WRAPPING_ALGOS = {"RSA/ECB/OAEPWithSHA-256AndMGF1Padding", "RSA/ECB/PKCS1Padding"};
 
-    private final JceMasterKey masterKey;
+    private JceMasterKey masterKey;
 
     private RSAMasterKey(final PublicKey publicKey,
                          final PrivateKey privateKey) throws SecretCryptoException {
         String keyId = publicKeySHA(publicKey);
-        this.masterKey = JceMasterKey.getInstance(publicKey, privateKey, KEY_PROVIDER, keyId, WRAPPING_ALGO);
+        Exception exception = null;
+        for (String algo : WRAPPING_ALGOS) {
+            try {
+                JceMasterKey possibleKey = JceMasterKey.getInstance(publicKey, privateKey, KEY_PROVIDER, keyId, algo);
+                // Force the usage of the algorithm to make sure it is supported. If it isn't supported we'll try the
+                // next one in line.
+                AwsCrypto.standard().encryptData(possibleKey, new byte[0]);
+                this.masterKey = possibleKey;
+                return;
+            } catch (Exception e) {
+                exception = e;
```

Comment:
Need to log this?

Reply:
It is logged just below. We don't need to log every exception when we find an algorithm that works.